apps/ios/SimpleX SE/ShareViewController.swift

```diff
@@ -32,3 +32,22 @@ class ShareViewController: UIHostingController<ShareView> {
         }
     }
 }
+
+// TODO: Database access management
+/// Handle database access using existing (not-ideal) model, since
+///
+/// 1. Subscribe to lifecycle notifications:
+///     - `NSExtensionHostDidBecomeActive`
+///     - `NSExtensionHostWillResignActive`
+///     - `NSExtensionHostDidEnterBackground`
+///     - `NSExtensionHostWillEnterForeground`
+///
+/// 2. Add `SEState` using `SharedFileSubscriber`
+/// and suspend the chat in case there is a NSE Notification, while share sheet process is active.
```

Comment:
you mean suspend SE chat, right?

Reply:
Yes - so that the notification can be delivered.
Not entirely sure of the implications of this though.

Reply:
it's not needed here, as we never enabled subscriptions in the first place, and if we close the database it won't do anything... It might still continue trying to deliver messages though next time apple wakes it up... So maybe we should stop it instead of suspending. need to think

Reply:
maybe we should suspend indeed... and then foreground it if it is opened in the same process.